superset-frontend/src/components/ReportModal/index.tsx

```diff
@@ -191,13 +160,11 @@ const ReportModal: FunctionComponent<ReportProps> = ({
   const [cronError, setCronError] = useState<CronError>();
   const dispatch = useDispatch();
   // Report fetch logic
-  const reports = useSelector<any, AlertObject>(state => state.reports);
-  const isEditMode = reports && Object.keys(reports).length;
+  const report = findReport;
```

Comment:
instead of passing in the report, I would suggest fetching from Redux as much as possible. You can import the function that you pass to useSelector if you want to reuse the code. 
```
const report = useSelector<any, AlertObject>(reportSelector)
```